### PR TITLE
feat: integration tests in ci

### PR DESCRIPTION
**Pull request by mersinvald**
Depends on #36 
Depends on #39 
Depends on #40 
Depends on #41 
Depends on #42

### CODE REVIEW

**Comment by mersinvald** [comment]
bors try

File: cleanroom/src/command/packing.rs [hunk 1]
```diff
@@ -6,43 +6,27 @@ use std::path::{Path, PathBuf};
...
-impl Unpack {
-    pub fn new() -> Self {
-        Unpack {}
-    }
-}
-
 impl CommandExecutor for Pack {
     type Ctx = PathBuf;
     fn execute(self, ctx: &Self::Ctx) -> anyhow::Result<()> {
+        crate::init_logger(None, None);
```
Comment (AnthonyMikh) [inline cleanroom/src/command/packing.rs]: This call looks weird. Maybe introduce `init_default_logger`?

File: cleanroom/src/test_runner/mod.rs [hunk 2]
```diff
@@ -122,11 +119,21 @@ impl TestRunner<Processed> {
             .diff_index_to_index(self.0.old_index(), self.0.new_index(), None)?;
         let mut new_diff = String::new();
         diff.print(DiffFormat::Patch, |_delta, _hunk, line| {
+            let contents = str::from_utf8(line.content()).unwrap();
+
             match line.origin() {
                 '+' | '-' | ' ' => new_diff.push(line.origin()),
```
Comment (AnthonyMikh) [inline cleanroom/src/test_runner/mod.rs]:
```suggestion
                orig@'+' | orig@'-' | orig@' ' => new_diff.push(orig),
```

File: cleanroom/src/test_runner/mod.rs [hunk 3]
```diff
@@ -122,11 +119,21 @@ impl TestRunner<Processed> {
...
                 _ => {}
             }
-            new_diff.push_str(str::from_utf8(line.content()).unwrap());
+
+            for line in contents.lines() {
+                if line.starts_with("index") {
```
Comment (AnthonyMikh) [inline cleanroom/src/test_runner/mod.rs]: Are you sure that this one will not cause false negatives?

File: cleanroom/src/test_runner/state.rs [hunk 4]
```diff
@@ -20,13 +18,12 @@ pub struct InitialToPrepared {
     pub index: Index,
 }
 
-impl<'a> Progress for Initial<'a> {
-    type Target = Prepared<'a>;
+impl<'a> Progress for Initial {
```
Comment (AnthonyMikh) [inline cleanroom/src/test_runner/state.rs]: Lifetime from `impl` generic parameter list is not used here. Remove it.

File: cleanroom/src/test_runner/state.rs [hunk 5]
```diff
@@ -51,7 +46,7 @@ pub struct PreparedIntoProcessed {
     pub index: Index,
 }
 
-impl<'a> Progress for Prepared<'a> {
+impl<'a> Progress for Prepared {
```
Comment (AnthonyMikh) [inline cleanroom/src/test_runner/state.rs]: Same here

File: cleanroom/src/command/packing.rs (hunk 1 above)
Comment (mersinvald) [reply cleanroom/src/command/packing.rs]: @AnthonyMikh how about `init_logger` and `init_logger_with`?

File: cleanroom/src/test_runner/mod.rs (hunk 3 above)
Comment (mersinvald) [reply cleanroom/src/test_runner/mod.rs]: I don't think it's critical for the use case. Most of the data would be json anyway.

File: cleanroom/src/test_runner/state.rs (hunk 4 above)
Comment (mersinvald) [reply cleanroom/src/test_runner/state.rs]: Oops, merge mistake.

**Pull request merged**